modules/core/src/shaderlib/project/viewport-uniforms.js

```diff
@@ -35,11 +37,38 @@ const IDENTITY_MATRIX = [1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1];
 const DEFAULT_PIXELS_PER_UNIT2 = [0, 0, 0];
 const DEFAULT_COORDINATE_ORIGIN = [0, 0, 0];
 
-// TODO: Find the best value for this to maximize accuracy
-const LNGLAT_EXPERIMENTAL_ZOOM_THRESHOLD = 12;
+// Based on viewport-mercator-project/test/fp32-limits.js
+const LNGLAT_AUTO_OFFSET_ZOOM_THRESHOLD = 12;
 
 const getMemoizedViewportUniforms = memoize(calculateViewportUniforms);
 
+function getShaderCoordinateSystem(coordinateSystem, fp64) {
+  if (fp64) {
+    // This is the only mode that works with fp64
+    return PROJECT_COORDINATE_SYSTEM.LNG_LAT;
```

Comment:
Drop underscore. LNGLAT

Reply:
Breaks stuff, because the module uses `COORDINATE_SYSTEM_LNG_LAT`

Reply:
> Breaks stuff, because the GLSL uses COORDINATE_SYSTEM_LNG_LAT

Yes we'd need to change the GLSL. That is kind of the point of this separation. We can change the PROJECT_... constants freely as long as we maintain the user facing `COORDINATE_SYSTEM`.

Reply:
experimental-layers (mesh layer) references this, so it breaks things.